spec/controllers/crosswalk_issues_controller_spec.rb

```diff
@@ -0,0 +1,166 @@
+      end
+
+      it 'correctly updates Crosswalk fields from weams' do
+        post(:resolve_partial, params: params)
+        expect(crosswalk[:facility_code]).to eq(weam[:facility_code])
+        expect(crosswalk[:institution]).to eq(weam[:institution])
+        expect(crosswalk[:city]).to eq(weam[:city])
+        expect(crosswalk[:state]).to eq(weam[:state])
+      end
+    end
+
+    context 'when related Crosswalk exists' do
+      let(:issue) { create :crosswalk_issue, :partial_match_type, :with_weam_match, :with_crosswalk_match }
+      let(:weam) { CrosswalkIssue.find(params[:id]).weam }
+      let(:crosswalk) { CrosswalkIssue.find(params[:id]).crosswalk }
+
+      let(:params) do
+        {
+          id: issue.id,
+          cross: '44445555',
+          ope: '333666',
+          notes: 'test'
+        }
+      end
+
+      it 'does not update crosswalk_id' do
+        expect { post(:resolve_partial, params: params) }.not_to change { issue[:crosswalk_id] }
+      end
+
+      it 'correctly updates Crosswalk fields from params' do
+        post(:resolve_partial, params: params)
+        expect(crosswalk[:cross]).to eq(params[:cross])
+        expect(crosswalk[:ope]).to eq(params[:ope])
+        expect(crosswalk[:notes]).to eq(params[:notes])
+      end
+
+      it 'correctly updates Crosswalk fields from weams' do
+        post(:resolve_partial, params: params)
+        expect(crosswalk[:facility_code]).to eq(weam[:facility_code])
+        expect(crosswalk[:institution]).to eq(weam[:institution])
+        expect(crosswalk[:city]).to eq(weam[:city])
+        expect(crosswalk[:state]).to eq(weam[:state])
+      end
+    end
+
+    context 'when Crosswalk is resolved' do
+      let(:issue) { create :crosswalk_issue, :partial_match_type, :with_weam_match, :with_ipeds_hd_match }
+
+      let(:params) do
+        {
+          id: issue.id,
+          cross: issue.weam[:cross],
+          ope: issue.weam[:ope]
+        }
+      end
+
+      it 'is deleted' do
+        post(:resolve_partial, params: params)
+        expect(CrosswalkIssue.exists?(issue.id)).to eq(false)
+      end
+    end
+
+    context 'when Crosswalk is not resolved' do
+      let(:issue) { create :crosswalk_issue, :partial_match_type, :with_weam_match }
+
+      let(:params) do
+        {
+          id: issue.id,
+          cross: issue.weam[:cross],
+          ope: issue.weam[:ope]
+        }
+      end
+
+      it 'is not deleted' do
+        post(:resolve_partial, params: params)
+        expect(CrosswalkIssue.exists?(issue.id)).to eq(true)
+      end
+    end
+  end
+
+  describe 'GET #orphans' do
+    login_user
+
+    before do
+      create_list :crosswalk_issue, 3, :partial_match_type
+      create_list :crosswalk_issue, 2, :ipeds_orphan_type
+      get(:orphans)
+    end
+
+    it 'populates an array of crosswalk issues' do
+      expect(assigns(:issues).length).to eq(CrosswalkIssue.by_issue_type(CrosswalkIssue::IPEDS_ORPHAN_TYPE).count)
```

Comment:
Seems like you should be expecting the particular value here to make sure everything IS right. 0=0 is true, but might not be the numbers you're looking for if anything is off.

Reply:
Thanks!